Frameworks/include/CppWinRTHelpers.h

```diff
@@ -155,4 +156,32 @@ winrt::Windows::Foundation::IReference<float> optional<float>(const float& f) {
     return ref.as<winrt::Windows::Foundation::IReference<float>>();
 }
 
+//
+// Callback wrapper
+//
+namespace impl {
+    template <typename TCallable>
+    class callback {
+    public:
+        callback(const TCallable& c) : _callback(c) { }
+        callback(callback&& other) : _callback(std::move(other._callback)) { }
+
+        // Make sure all callback operations are wrapped in an autorelease pool
+        template <typename... TArgs>
+        auto operator()(TArgs&& ...args) {
+            @autoreleasepool {
+                return _callback(std::forward<TArgs>(args)...);
+            }
+        }
+
+    private:
+        const TCallable& _callback;
```

Comment:
(since the scope they close over will disappear shortly after they're registered via `objcwinrt::callback`.